File: Очереди с приоритетами/first.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

vector<int> split(const string &s, char delim) {
  vector<int> elems;
  stringstream ss;
  ss.str(s);
  string item;
  while (getline(ss, item, delim)) {
    elems.push_back(stoi(item));
  }
  return elems;
}

bool is_complete(string H){
  vector<int> elems = split(H, ',');
  for(auto e : elems){
    cout << e << ", ";
  }
  cout << endl;

  for(int i = 0; i < (elems.size() - 2) / 2; i++){
    if(!(elems[i] <= elems[2 * i + 1])) return false;
    if(!(elems[i] <= elems[2 * i + 2])) return false;
  }

  return true;
}

int main() {
  string s = "1, 4, 2, 3, 4, 5, 6";
  cout << boolalpha << is_complete(s);
}

File: Очереди с приоритетами/first.py
def is_complete(H):
  arr = H.split(',')
  arr = list(map(int, arr))

  print(arr)
  
  for i in range(0, (len(arr) - 2) // 2):
    if not (arr[i] <= arr[2 * i + 1]):
      return False
    if not (arr[i] <= arr[2 * i + 2]):
      return False

  return True

print(is_complete('1, 4, 2, 3, 4, 5, 6'))